tree/treeplayer/src/TTreeFormula.cxx

```diff
@@ -2352,7 +2352,7 @@ Int_t TTreeFormula::FindLeafForExpression(const char* expression, TLeaf*& leaf,
             final = true;
 
             // Record in 'i' what we consumed
-            i += strlen(params);
+            i += 2; // open and close parentheses
```

Comment:
That does not 'sound' sufficient unless 'i' is incremented by the size of the params content somewhere else, is it?

Reply:
I don't really follow the logic in this function, but this was needed in order to make the check at line 2619 work. So it was an ad-hoc solution.

Another way of solving this would be to record strlen(params) in an auxiliary variable and then do:

`if (i < nchname + paramLength - 2)`

would that be better?

Reply:
I checked to do this before my patch, and I get:

`mc.Get(1)->x` and I get 9 < 10
`mc.Get(12)->x` and I get 12 < 11
`mc.Get(123)->x` and I get 14 < 12

so the difference is increasing a factor of 2 per char. So it seems to me that without my patch, length(params) is being summed twice?

Reply:
Humm .. you are right, `i` is already pointing to the closing parenthesis at that point, so actually it feels like:
```
i += 0; // nothing to do really
```
seems like the 'right' number (i.e. removing the line entirely) ... but the fact that it sort of work sometime with 2 or some other random number is puzzling :)